homeassistant/components/recollect_waste/__init__.py

```diff
@@ -31,14 +32,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     async def async_get_pickup_events() -> list[PickupEvent]:
         """Get the next pickup."""
         try:
-            return await client.async_get_pickup_events(
+            data = await client.async_get_pickup_events(
                 start_date=date.today(), end_date=date.today() + timedelta(weeks=4)
             )
         except RecollectError as err:
             raise UpdateFailed(
                 f"Error while requesting data from ReCollect: {err}"
             ) from err
 
+        return cast(List[PickupEvent], data)
```

Comment:
`async_get_pickup_events` is already typed upstream, why would it need casting?

Reply:
Without this explicit casting, `mypy` complains:

```
homeassistant/components/recollect_waste/__init__.py:43: error: Returning Any from function declared to return "List[Any]"  [no-any-return]
```

Reply:
That sounds odd, as it is defined as `List[PickupEvent]` upstream

Reply:
Totally. I don't understand `mypy`. 😞

Reply:
Add a file called `py.typed` to your upstream package, to indicate its typed.

![image](https://user-images.githubusercontent.com/195327/127066385-648e71bb-0cb4-43e3-a0f9-384c70be8041.png)

Next:

![image](https://user-images.githubusercontent.com/195327/127066415-fb2be87f-f2ec-4f37-a686-0f7ff6b59d4d.png)


Just works.

![image](https://user-images.githubusercontent.com/195327/127066427-54e54525-5048-4aca-8eff-a05061005587.png)